src/packet_macros.rs

```diff
@@ -51,33 +51,51 @@ macro_rules! packet_boilerplate {
             stream.write_all(&byte_vec).unwrap();
         }
 
+        pub fn translate(packet: Packet, translation_info: TranslationInfo) -> Packet {
+            match packet {
+                $(Packet::$name(packet) => {
+                    Packet::$name(packet.translate(translation_info))
+                })*
+                Packet::Unknown => { Packet::Unknown }
+            }
+        }
+
         //Define the packet struct
-        $(packet!{$name, $id, [ $( ($fieldname, $datatype)),*]})*
+        $(packet!{$name, $id, [ $( ($fieldname, $datatype, $transtype)),*]})*
     )
 }
 
-//Example: packet!(0, Handshake, 0, [ (somefield, VarInt) ] becomes:
-//pub struct Handshake { somefield: u64 }
-//impl Handshake {
-//  pub fn new(stream) -> Handshake {
-//    Handshake { somefiled: stream.read_var_int() }
-//  }
-//  pub fn write(self, stream) -> Handshake {
-//    stream.write_var_int(self.somefield)
-//  }
-//}
 macro_rules! packet {
-    ($name:ident, $id:expr, [ $( ($fieldname:ident, $datatype:ident)),* ]) => (
+    ($name:ident, $id:expr, [ $( ($fieldname:ident, $datatype:ident, $transtype:ident)),+]) => (
```

Comment:
separated out the two cases here- packets with fields and packets without. This is mostly just for style, otherwise the generated code gets real ugly and sets off clippy. So I did this instead of throwing in a bunch of style disablers